AGENTS_MIGRATION.md

```diff
@@ -0,0 +1,164 @@
+
+### 2. Multiple Agent Types (ServiceExtensions.cs)
+```csharp
+// EXTENSION POINT: Future multi-agent support could register additional agents here
+// For example:
+// services.AddTransient<NPCAgent>(sp => CreateNPCAgent(sp, "Merchant"));
+// services.AddTransient<WorldAgent>(sp => CreateWorldAgent(sp));
+```
+
+**Suggested Agent Types:**
+- **NPCAgent**: Roleplay specific non-player characters
+- **WorldAgent**: Provide environmental descriptions and world state
+- **RulesAgent**: Handle rule enforcement and dice rolling
+- **NarratorAgent**: Focus on storytelling and scene setting
+
+### 3. Agent Configuration (GameMasterAgentFactory.cs)
+```csharp
+// EXTENSION POINT: Agent configuration could be enhanced for multi-agent scenarios
+// For example, different agent types could have different configurations:
+// - GameMasterAgent: Narrative control, rule enforcement
+// - NPCAgent: Character roleplay for specific NPCs
+// - WorldAgent: Environmental descriptions and world state
+```
+
+## Implementation Example: Adding NPC Agents
+
+### Step 1: Create NPC Agent Factory
+```csharp
+public class NPCAgentFactory
+{
+    public ChatCompletionAgent CreateNPCAgent(Character npc, Adventure adventure, Kernel kernel)
+    {
+        return new ChatCompletionAgent
+        {
+            Name = $"NPC_{npc.Name}",
+            Description = $"Roleplay agent for {npc.Name}",
+            Instructions = $"You are {npc.Name}, a {npc.Specialization}. {npc.CharacterSheet}",
+            Kernel = kernel
+        };
+    }
+}
+```
+
+### Step 2: Register Multiple Agents
+```csharp
+services.AddTransient<NPCAgentFactory>();
+services.AddTransient<IEnumerable<Agent>>(sp =>
+{
+    var adventure = sp.GetRequiredService<Adventure>();
+    var kernel = sp.GetRequiredService<Kernel>();
+    var npcFactory = sp.GetRequiredService<NPCAgentFactory>();
+    var gmFactory = sp.GetRequiredService<GameMasterAgentFactory>();
+    
+    var agents = new List<Agent>
+    {
+        gmFactory.CreateGameMasterAgent(adventure, character, kernel)
+    };
+    
+    // Add NPC agents for each character in the adventure
+    agents.AddRange(adventure.Characters.Select(npc => 
+        npcFactory.CreateNPCAgent(npc, adventure, kernel)));
+    
+    return agents;
+});
+```
+
+### Step 3: Implement Agent Group Coordination
+```csharp
+public class MultiAgentChatClient : IAgentChatClient
+{
+    private readonly AgentGroupChat _agentGroup;
+    
+    public MultiAgentChatClient(IEnumerable<Agent> agents)
+    {
+        _agentGroup = new AgentGroupChat(agents.ToArray());
+    }
+    
+    public async Task<ChatHistory> ChatAsync(ChatHistory history, CancellationToken cancellationToken)
+    {
+        await foreach (var message in _agentGroup.InvokeAsync(history, cancellationToken))
+        {
+            history.Add(message);
+            // Display message with agent identification
+        }
+        return history;
+    }
+}
+```
+
+## Backward Compatibility
```

Comment:
Backward compatibility is not needed. Your design should be optimal for agents framework and not need to worry about significant changes in signatures.